app/src/main/res/values/strings.xml

```diff
@@ -99,6 +99,11 @@
     <string name="action_login">Login with Mastodon</string>
     <string name="action_logout">Log Out</string>
     <string name="action_logout_confirm">Are you sure you want to log out of the account %1$s?</string>
+    <string name="action_post_failed">Upload failed</string>
+    <string name="action_post_failed_detail">A post failed to upload and has been saved to drafts.</string>
+    <string name="action_post_failed_detail_plural">%1$d posts failed to upload and have been saved to drafts.</string>
```

Comment:
Very unsure about any of this wording. It does not seem "friendly".

The number of uploaded posts is totally superfluous (and according to Google, including the number will cause [problems for, at least, Lithuanian?](https://developer.android.com/guide/topics/resources/string-resource#Plurals)), but I thought it would be nice to display a different message for one failed post vs more-than-one.

Currently thinking about changing it to "Your post failed to upload and has been saved to drafts." / "Your posts failed to upload and have been saved to drafts." and dropping the number.

Also not sure about the use of "Upload" but I can't think of a better word.

Suggestions welcomed.

Reply:
"Your Mastodon server temporarily rejected your post. It has been saved as a draft so you can try again."

- Makes it clear that this is the server's fault, not the user's (and not the app's)
- Makes it clear that this is a temporary issue
- Makes it clear they they can (and should) try again

In the actual dialog that pops up you could also include the text of the post, so the user can see which post is being referred to.

If multiple posts fail you can show multiple dialogs. That wouldn't be great UX (the user might have to click through multiple dialogs in the worst cast), but I suspect that multiple posts failing would be a rare occurrence, so probably not a problem (?)